ergotree-ir/src/mir/coll_append.rs

```diff
@@ -0,0 +1,124 @@
+use crate::serialization::op_code::OpCode;
+use crate::serialization::sigma_byte_reader::SigmaByteRead;
+use crate::serialization::sigma_byte_writer::SigmaByteWrite;
+use crate::serialization::SerializationError;
+use crate::serialization::SigmaSerializable;
+// use crate::types::stuple::STuple;
+use crate::types::stype::SType;
+
+
+use super::expr::Expr;
+use super::expr::InvalidArgumentError;
+use crate::has_opcode::HasStaticOpCode;
+
+/// Takes two collections as input and concatenates both into the Append Colelction
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub struct Append {
+    /// Collection - First Parameter; first half of the combined collection
+    pub input: Box<Expr>,
+    /// Collection - Second Parameter; later half of the combined collection
+    pub output: Box<Expr>,
```

Comment:
```suggestion
    pub col2: Box<Expr>,
```
I find this name confusing. It's not "output" in a sense of this being the result of the op. It's the "second" collection that will be appended to the target collection.